File: jvm2/src/Model/Abstract/ITile.kt
package Model.Abstract

import Model.Location
import Model.Move

interface ITile {
    var loc: Location
    var occ: Boolean
    var use: Boolean
    var drive: Boolean
    var traverse: Boolean
    var owner: Int
    var blank: Boolean
    var minCost : Int
    var gasCost : Int
    var nrgCost : Int
    var minSal : Int
    var gasSal : Int
    var nrgSal : Int
    var wait : Int
    var health : Int
    var abort : Boolean
    var commands : MutableList<String>
    var icon: String
    fun getPNG() : String
}

File: dotnet/src/Model/Abstract/ITile.cs
using System;
using System.Collections.Generic;

namespace PlanetCS1 {
    public interface ITile {
        Location loc {get;set;}
        Boolean occ {get;set;}
        Boolean use {get;set;}
        Boolean drive {get;set;}
        Boolean traverse {get;set;}
        int owner {get;set;}
        Boolean blank {get;set;}
        int minCost {get;set;}
        int gasCost {get;set;}
        int nrgCost {get;set;}
        int minSal {get;set;}
        int gasSal {get;set;}
        int nrgSal {get;set;}
        int wait {get;set;}
        int health {get;set;}
        Boolean abort {get;set;}
        List<String> commands {get;set;}
        String icon {get;set;}
    }
}